src/components/Pagination.astro

```diff
@@ -1,49 +1,43 @@
 ---
+import type { Page } from "astro";
 import LinkButton from "./LinkButton.astro";
+import type { CollectionEntry } from "astro:content";
 
 export interface Props {
-  currentPage: number;
-  totalPages: number;
-  prevUrl: string;
-  nextUrl: string;
+  page: Page<CollectionEntry<"blog">>;
 }
 
-const { currentPage, totalPages, prevUrl, nextUrl } = Astro.props;
-
-const prev = currentPage > 1 ? "" : "disabled";
-const next = currentPage < totalPages ? "" : "disabled";
-const isPrevDisabled = prev === "disabled";
-const isNextDisabled = next === "disabled";
+const { page } = Astro.props;
 ---
 
 {
-  totalPages > 1 && (
+  page.lastPage > 1 && (
     <nav class="pagination-wrapper" aria-label="Pagination">
       <LinkButton
-        disabled={isPrevDisabled}
-        href={prevUrl}
-        className={`mr-4 select-none ${prev}`}
+        disabled={!page.url.prev}
+        href={page.url.prev as string}
+        className={`mr-4 select-none`}
```

Comment:
I noticed something.
You also need to conditionally add `.disabled` class in the prev and next buttons.
Otherwise, disabled button isn't distinguishable visually.

![image](https://github.com/user-attachments/assets/3fb9aca2-6e5b-45b8-8473-35ba21eb6d9d)